core/src/main/java/tc/oc/pgm/community/modules/FreezeMatchModule.java

```diff
@@ -369,14 +368,27 @@ private void thaw(MatchPlayer freezee, Component senderName, boolean silent) {
       freezee.sendMessage(thawedTitle.color(TextColor.GREEN).build());
 
       ChatDispatcher.broadcastAdminChatMessage(
-          TranslatableComponent.of(
-              "moderation.freeze.broadcast.thaw",
-              TextColor.GRAY,
-              senderName,
-              freezee.getName(NameStyle.CONCISE)),
+          createInteractiveBroadcast(
+              freezee.getName(NameStyle.CONCISE), senderName, freezee.getBukkit().getName(), false),
           match);
     }
 
+    private Component createInteractiveBroadcast(
+        Component senderName, Component targetName, String username, boolean frozen) {
+      return TextComponent.builder()
```

Comment:
Is this mixed up?
Constuctor calls for `senderName`, `targetName`. Seems like you are sending `targetName` and then `senderName` rn
Also considered just having a constructor `MatchPlayer sender, MatchPlayer target, boolean frozen` ?

so a method `Component createInteractiveFreezeBroadcast` could be something like
```java
 return TextComponent.builder()
.append(
              TranslatableComponent.of(
                  String.format("moderation.freeze.broadcast.%s", frozen ? "frozen" : "thaw"),
                  TextColor.GRAY,
                  sender.getName(NameStyle.CONCISE),
                  target.getName(NameStyle.CONCISE))
          .hoverEvent(
              HoverEvent.showText(
                  TranslatableComponent.of("moderation.freeze.broadcast.hover", TextColor.GRAY)))
          .clickEvent(ClickEvent.runCommand("/f " + target.getBukkit().getName())
          .build();
```
The thing i think is weird is passing a `targetName` and a `username`

Reply:
Hey Simon!
Thanks for the feedback, I see what I did xD I'll be making some changes later in the day to make this method more precise. Thanks!